### ISSUE TITLE
.NET 2.0 is not obsolete!

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/windows-package-manager#85]
```
Visual C++ & .NET Versions must be independent and the word "obsolete" must be 
removed. Some programs need especifical versions and we must have all installed 
at the same time.

Please, remove the word obsolete and check all versions install.
```

Original issue reported on code.google.com by `pr1nc...@gmail.com` on 18 Jul 2011 at 6:35

**Comment by GoogleCodeExporter** [comment]
```
isn't for example .NET framework 2.0 obsolete? Many programs also require "at 
least .NET 2.0" and will also work with 4.0.

```

Original comment by `tim.lebe...@gmail.com` on 18 Jul 2011 at 7:35

**Comment by GoogleCodeExporter** [comment]
```
Yes, but some bad developers and aplications work bad if you dont have install 
especifical versions. Also some software if not detect old versions install it.
```

Original comment by `pr1nc...@gmail.com` on 18 Jul 2011 at 10:50

**Comment by GoogleCodeExporter** [comment]
```
Let me try to describe the problem with my own words.
Suppose a user installed without Npackd the following:
- .NET framework 2.0
- .NET framework 3.5
- a program P based on .NET framework 2.0 that is incompatible with .NET 3.5, 
but claims so
- Npackd

If this user starts Npackd, he sees that .NET framework 2.0 is obsolete. The 
user may decide to uninstall .NET framework 2.0 through the control panel 
because of this. This would make the program P unusable.

Is this right?
```

Original comment by `tim.lebe...@gmail.com` on 19 Jul 2011 at 2:48

**Comment by GoogleCodeExporter** [comment]
```
Perfect! ALL versions of.NET & Visual C++ can be instaled at the same time and 
NOT are obsolete.
```

Original comment by `pr1nc...@gmail.com` on 23 Jul 2011 at 7:17

**Comment by GoogleCodeExporter** [comment]
Original comment by `tim.lebe...@gmail.com` on 23 Jul 2011 at 9:08
* Changed state: **Thinking**

**Comment by GoogleCodeExporter** [comment]
```
If you would describe a dependency on .NET in a repository, Npackd would not 
allow you to un-install it. What applications specifically are we talking here 
about?
```

Original comment by `tim.lebe...@gmail.com` on 31 Jul 2011 at 9:22

**Comment by GoogleCodeExporter** [comment]
Original comment by `tim.lebe...@gmail.com` on 9 Aug 2011 at 8:45
* Added labels: **Type-Enhancement**
* Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Games and old stuff that re-instalate undetected specifically versions.
```

Original comment by `pr1nc...@gmail.com` on 19 Aug 2011 at 1:06

**Comment by GoogleCodeExporter** [comment]
```
I will not split .NET 2, 3 and 4 into independent packages. They are clearly 
the same product/package. Nonetheless it would be interesting to know the 
differences between 2 and 3 or 4 and 4.5. It could be useful to mark some 
versions as "only-bugfixes" or "compatible" or "security-related". This could 
also change the descriptions for those packages from "obsolete" to something 
else.
```

Original comment by `tim.lebe...@gmail.com` on 10 Sep 2011 at 9:59
* Changed title: **Add the purpose of a package version to the list of attributes (e.g. "security-related" or "compatible")**

**Comment by GoogleCodeExporter** [comment]
```
Idea - Make a "don't follow" status for specific software that we want to admin 
externally npack!
```

Original comment by `ismael.n...@gmail.com` on 1 Oct 2011 at 3:37

**Comment by GoogleCodeExporter** [comment]
Original comment by `tim.lebe...@gmail.com` on 15 Oct 2011 at 7:00

**Comment by GoogleCodeExporter** [comment]
```
1. What are you basing the notion that .Net 2, 3, and 4 are the same product 
and not different products with a similar name?
2. Its not entirely true to say that there are no obsolete versions. There are 
micro updates. This is similar to java where 1.6u33 makes 1.6 through 1.6u32 
obsolete but 1.7 and 1.5 are not. Separate branches vs micro updates.
3. It is not a matter of instability, if you install VC++ or .Net of the wrong 
branch then applications outright don't work. I have tried and failed to use 
older programs use newer versions and it just doesn't work. It is simply 
ESSENTIAL that this allows installing all the various branches.
4. All of them are concurrently installed in newer windows versions (ex: Win7) 
and cannot be uninstalled, only updated.
```

Original comment by `MrT...@gmail.com` on 2 Aug 2012 at 1:04

**Comment by GoogleCodeExporter** [comment]
```
Changing the title back...

1.17 has some UI changes. Please take a look at the attached screenshots. The 
word "obsolete" will only be visible on tabs for particular package versions 
and only if the newest package version is installed.

This does not change the logic for marking package versions as "obsolete" 
though. 
```

Original comment by `tim.lebe...@gmail.com` on 3 Aug 2012 at 6:01
* Changed title: **.NET 2.0 is not obsolete!**

Attachments:
* [Overview.png](https://storage.googleapis.com/google-code-attachments/windows-package-manager/issue-85/comment-13/Overview.png)
* [Package.png](https://storage.googleapis.com/google-code-attachments/windows-package-manager/issue-85/comment-13/Package.png)
* [PackageVersion.png](https://storage.googleapis.com/google-code-attachments/windows-package-manager/issue-85/comment-13/PackageVersion.png)

**Comment by GoogleCodeExporter** [comment]
```
And now the answers to MrTTAO questions:

1. 
- the *same* name: .NET
- the same vendor: Microsoft
- many programs depend on .NET with *at least* version X
- increasing version numbers

2. I think JDK 1.5 is obsolete as Windows XP is obsolete. Which word instead of 
"obsolete" would you prefer?

3. agree

4. you can uninstall .NET 2.0: http://support.microsoft.com/kb/908077

```

Original comment by `tim.lebe...@gmail.com` on 3 Aug 2012 at 6:13

**Comment by GoogleCodeExporter** [comment]
```
1. And yet, they are different product with similar names.
.Net 1 is a different product then .Net 2. Despite having the same first half 
of the name, the same vendor, and programs requiring at least versions, and 
increasing numbers.

2. I agree about it being as obsolete as XP... that is to say, its old, its 
crummy, and it MUST be used by specific ancient programs that were never 
properly updated.
I encounter enterprise products that work only on XP on a regular basis.

4. Interesting.

Overall:
I think that for those kind of programs we should, through manual consideration 
on a per program basis, declare separate branches that are designed to be 
installed concurrently. And assign updates within those branches when 
appropriate.
```

Original comment by `MrT...@gmail.com` on 11 Aug 2012 at 8:58

**Comment by GoogleCodeExporter** [comment]
```
BTW, Microsoft removed the .NET 2.0 redistributable from their download site. 
Does anybody know where to get it (a stable URL)?
```

Original comment by `tim.lebe...@gmail.com` on 25 Aug 2012 at 4:04

**Comment by GoogleCodeExporter** [comment]
```
this issue became some sort of a discussion. I think the initial problem is 
solved at least partially in Npackd 1.17. Please file another specific issues 
if required.
```

Original comment by `tim.lebe...@gmail.com` on 6 Sep 2012 at 7:01
* Changed state: **Fixed**
* Added labels: **Milestone-1.17**
